### PR TITLE
189 User video functionality

### PR DESCRIPTION
**Pull request by ayomanetsan**
This PR features a fully updated video functionality. In a nutshell that is
1. The enhanced look of the call page itself;
2. Real-time notifications;
3. Database saving.

### CODE REVIEW

File: backend/EasySpeak.Communicator/EasySpeak.Communicator.WebAPI/EasySpeak.Communicator.WebAPI.csproj [hunk 1]
```diff
@@ -13,4 +13,8 @@
     <PackageReference Include="Swashbuckle.AspNetCore" Version="6.2.3" />
   </ItemGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="..\..\EasySpeak.Core\EasySpeak.Core.DAL\EasySpeak.Core.DAL.csproj" />
+  </ItemGroup>
+
```
Comment (tatianahutii) [inline backend/EasySpeak.Communicator/EasySpeak.Communicator.WebAPI/EasySpeak.Communicator.WebAPI.csproj:19]: dockerfile?

File: backend/EasySpeak.Communicator/EasySpeak.Communicator.WebAPI/Services/SignalRtcHub.cs [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
 
 namespace EasySpeak.Communicator.WebAPI.Services
 {
     public class SignalRtcHub : Hub
     {
+        private readonly EasySpeakCoreContext _context;
         private static readonly Dictionary<string, string> ConnectedUsers = new Dictionary<string, string>();
         static readonly Dictionary<string, List<string>> ConnectedClients = new Dictionary<string, List<string>>();
 
+        public SignalRtcHub(EasySpeakCoreContext context)
```
Comment (tatianahutii) [inline backend/EasySpeak.Communicator/EasySpeak.Communicator.WebAPI/Services/SignalRtcHub.cs:13]: rename file and class

File: frontend/src/app/core/hubs/webrtc-hub.service.ts [hunk 3]
```diff
@@ -84,8 +128,46 @@ export class WebrtcHubService {
         });
     }
 
-    public async callUser(email: string, roomName: string) {
-        await this.hubConnection.invoke('CallUser', email, roomName).catch((err) => console.error(err));
+    public async callUser(
```
Comment (tatianahutii) [inline frontend/src/app/core/hubs/webrtc-hub.service.ts:131]: interface for this

File: frontend/src/app/core/hubs/webrtc-hub.service.ts [hunk 4]
```diff
@@ -84,8 +128,46 @@ export class WebrtcHubService {
...
+        calleeEmail: string,
+        callerId: number,
+        callerEmail: string,
+        callerFullName: string,
+        callerImgPath: string,
+    ) {
+        await this.hubConnection
+            .invoke('CallUser', chatId, calleeEmail, callerId, callerEmail, callerFullName, callerImgPath)
+            .catch((err) => {
+                console.error(err);
```
Comment (tatianahutii) [inline frontend/src/app/core/hubs/webrtc-hub.service.ts:142]: remove console.error and show toaster

Comment (tatianahutii) [reply frontend/src/app/core/hubs/webrtc-hub.service.ts:142]: everywhere

File: frontend/src/app/core/hubs/hubFactories/webrtc-hub-factory.service.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { Injectable } from '@angular/core';
+import { SignalRHubFactoryService } from '@core/hubs/hubFactories/signalr-hub-factory.service';
+import { environment } from '@env/environment';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class WebrtcHubFactoryService extends SignalRHubFactoryService {
```
Comment (tatianahutii) [inline frontend/src/app/core/hubs/hubFactories/webrtc-hub-factory.service.ts:8]: did you check Nginx files for correct work with WebSockets?

**Review by tatianahutii — DISMISSED** [review]

**Review by ishootdice — APPROVED** [review]

**Pull request merged**